app/converters/sync_asset_segment_events_converter.rb

```diff
@@ -0,0 +1,27 @@
+module SyncAssetSegmentEventsConverter
+  include BodConverter
+
+  def process_sync_noun(noun)
+    begin
+      asset = Equipment.find_by uuid: noun.dig("assetSegmentEvent","asset","uUID")
+      floc = FunctionalLocation.find_by uuid: noun.dig("assetSegmentEvent","segment","uUID")
+
+      unless floc.present?
+        floc = create_missing_floc(noun.dig("assetSegmentEvent","segment"), "SyncAssetSegmentEvents")
+      end
+
+      unless asset.present?
+        asset = create_missing_asset(noun.dig("assetSegmentEvent","asset"), "SyncAssetSegmentEvents")
+      end
```

Comment:
You could potentially simplify these constructs by making 'create_missing_floc' to be more of an 'ensure_floc_exists' and then use `FunctionalLocation.find_or_initialize` or `FunctionalLocation.find_or_create` to do the lookup or creation in one step. That way you do not have to duplicate the find and check if it was found code as much.